### ISSUE TITLE
"View not found in react-native"

### ISSUE THREAD

**Issue by jchook** [import-js/eslint-plugin-import#1754]
Some modules do not resolve with eslint-plugin-import that DO resolve in react-native.

See https://github.com/facebook/react-native/issues/28549

I noticed that this also happens with [react-native-screens](https://github.com/software-mansion/react-native-screens).

> Unable to resolve path to module 'react-native-screens'.

When I look at `node_modules/react-native/index.js`, I can see some pretty "non-standard" commonJS exports, which I suspect could contribute to the issue, but otherwise I don't see why it would fail on these modules in particular.

```ts
module.exports = {
  // Components
  get AccessibilityInfo(): AccessibilityInfo {
    return require('./Libraries/Components/AccessibilityInfo/AccessibilityInfo');
  },
  get ActivityIndicator(): ActivityIndicator {
    return require('./Libraries/Components/ActivityIndicator/ActivityIndicator');
  },
  get Button(): Button {
    return require('./Libraries/Components/Button');
  },
...
```

For `react-native-screens`, it appears to use the (rarely used?) [files](https://docs.npmjs.com/files/package.json#files) directive in `package.json`.

Code that produces the false positive linter error:

```js
import { enableScreens } from 'react-native-screens' // error
import { View } from 'react-native' // error
```

**Comment by ljharb** [comment]
Indeed, using a getter on `module.exports` is both slow and causes all sorts of subtle issues - and also makes any linter plugin unable to statically resolve imports.

I'd suggest filing issues on react-native to stop using dynamically loaded exports. I'm not sure how else any linter could support it (and notably, native ESM in node absolutely doesn't).

**Comment by soryy708** [comment]
Out of scope for the plugin, looks like it should be addressed either by react-native or by ESLint
